FW/FreeRTOS/Task.hpp

```diff
@@ -0,0 +1,146 @@
+#ifndef _FREERTOS_TASK_HPP
+#define _FREERTOS_TASK_HPP 
+
+#include "FreeRTOS.h"
+#include "Task.h"
+#include "string.h"
+
+namespace FreeRTOS
+{
+    /**
+     * @brief FreeRTOS task wrapper
+     * @note Template stackSize have to be a multiple of sizeof(StackType_t)
+     * 
+     * @tparam stackSize 
+     */
+    template <size_t stackSize>
+    class Task {
+    public:
+
+        /**
+         * @brief Construct a new Task:: Task object
+         * 
+         * @param name Task name, will be truncated if longer than configMAX_TASK_LENGTH
+         * @param stackSize Stack size in bytes, rounded up to nearest sizeof(portSTACK_TYPE)
+         * @param priority Task priority
+         */
+        Task(const char *name, UBaseType_t priority) {
+            // TODO: assert(stackSize % sizeof(StackType_t));
+
+            strncpy(m_name, name, configMAX_TASK_NAME_LEN);
```

Comment:
As an aside, I realized later you could also do:
```c++
snprintf(m_name, configMAX_TASK_NAME_LEN, "%s", name);
```
since snprintf() guarantees null termination, but nevermind now, the code as written is perfectly fine =)